spec/lang/step/expressions.md

```diff
@@ -269,8 +269,8 @@ impl<M: Memory> Machine<M> {
         };
         assert!(offset <= ty.size::<M::T>());
 
-        let ptr = self.ptr_offset_inbounds(root.ptr, offset.bytes())?;
-        ret((Place { ptr, ..root }, field_ty))
+        let ptr = self.ptr_offset_inbounds(root.ptr.thin_pointer, offset.bytes())?;
+        ret((Place { ptr: ptr.widen(root.ptr.metadata), ..root }, field_ty))
```

Comment:
This is not in general correct -- only projections to the last field preserve metadata.

So maybe make this `None` for now and add a TODO.